src/server/services/wms/qgswmsdescribelayer.cpp

```diff
@@ -20,28 +20,53 @@
  ***************************************************************************/
 #include "qgswmsutils.h"
 #include "qgswmsdescribelayer.h"
-#include "qgswmsservertransitional.h"
 
 namespace QgsWms
 {
 
   void writeDescribeLayer( QgsServerInterface* serverIface, const QString& version,
                            const QgsServerRequest& request, QgsServerResponse& response )
   {
-    Q_UNUSED( version );
-    QgsServerRequest::Parameters params = request.parameters();
-
-    QgsWmsServer server( serverIface->configFilePath(),
-                         *serverIface->serverSettings(),
-                         params,
-                         getConfigParser( serverIface ),
-                         serverIface->accessControls() );
-    QDomDocument doc = server.describeLayer();
+    QDomDocument doc = describeLayer( serverIface, version, request );
     response.setHeader( QStringLiteral( "Content-Type" ), QStringLiteral( "text/xml; charset=utf-8" ) );
     response.write( doc.toByteArray() );
   }
 
+  // DescribeLayer is defined for WMS1.1.1/SLD1.0 and in WMS 1.3.0 SLD Extension
+  QDomDocument describeLayer( QgsServerInterface* serverIface, const QString& version,
+                              const QgsServerRequest& request )
+  {
+    Q_UNUSED( version );
+
+    QgsServerRequest::Parameters parameters = request.parameters();
+    QgsWmsConfigParser* configParser = getConfigParser( serverIface );
 
+    if ( !parameters.contains( QStringLiteral( "SLD_VERSION" ) ) )
```

Comment:
I read the post: the assertion is very vague and gives no hints about wich kind algorithm is impacted by the choice of one are another type: In our case it would be the computation of the order relation between two strings ( it's a QMap ). I would bet that the fastest cas would be when the compared strings are of the same type.